streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -59,28 +53,92 @@ private ClientState(Set<T> activeTasks, Set<T> standbyTasks, Set<T> assignedTask
     }
 
     public void assign(T taskId, boolean active) {
-        if (active)
+        if (active) {
             activeTasks.add(taskId);
-        else
+        } else {
             standbyTasks.add(taskId);
+        }
 
         assignedTasks.add(taskId);
+    }
+
+    public Set<T> activeTasks() {
+        return activeTasks;
+    }
+
+    public Set<T> standbyTasks() {
+        return standbyTasks;
+    }
 
-        double cost = COST_LOAD;
-        cost = prevAssignedTasks.remove(taskId) ? COST_STANDBY : cost;
-        cost = prevActiveTasks.remove(taskId) ? COST_ACTIVE : cost;
+    public int assignedTaskCount() {
+        return assignedTasks.size();
+    }
+
+    public void incrementCapacity() {
+        capacity++;
+    }
+
+    public int activeTaskCount() {
+        return activeTasks.size();
+    }
+
+    public void addPreviousActiveTasks(final Set<T> prevTasks) {
+        prevActiveTasks.addAll(prevTasks);
+        prevAssignedTasks.addAll(prevTasks);
+    }
 
-        this.cost += cost;
+    public void addPreviousStandbyTasks(final Set<T> standbyTasks) {
+        prevAssignedTasks.addAll(standbyTasks);
     }
 
     @Override
     public String toString() {
         return "[activeTasks: (" + activeTasks +
             ") assignedTasks: (" + assignedTasks +
+            ") standbyTasks: (" + standbyTasks +
             ") prevActiveTasks: (" + prevActiveTasks +
             ") prevAssignedTasks: (" + prevAssignedTasks +
             ") capacity: " + capacity +
-            " cost: " + cost +
             "]";
     }
+
+    boolean reachedCapacity() {
+        return assignedTasks.size() >= capacity;
```

Comment:
Capacity is the number of consumers, i.e. stream threads on this client; and one thread should be able to handle multiple tasks. Should we use this criterion that the assigned tasks has exceeded the number of threads to determine if the client is "full"?